src/modules/role/role.service.ts

```diff
@@ -205,40 +205,25 @@ export class RoleService {
     return verifiedRoles.filter(Boolean);
   }
 
-  public async verifyEnrolmentPrecondition({
-    userDID,
+  /**
+   * Fetches enrolment preconditions for a given role (claim type)
+   * @param claimType the role to fetch enrolment preconditions for
+   * @return enrolment preconditions for a given role (claim type)
+   */
+  public async fetchEnrolmentPreconditions({
     claimType,
   }: {
-    userDID: string;
     claimType: string;
   }) {
-    const [didDocument, role] = await Promise.all([
-      this.didService.getById(userDID),
-      this.getByNamespace(claimType),
-    ]);
-
+    const role = await this.getByNamespace(claimType);
     if (!role) {
       throw new Error(`There is no created role for ${claimType} namespace`);
     }
 
     const {
       definition: { enrolmentPreconditions },
     } = role;
-
-    if (!enrolmentPreconditions || enrolmentPreconditions.length < 1) return;
-    for (const { type, conditions } of enrolmentPreconditions) {
-      if (type === 'role' && conditions && conditions?.length > 0) {
-        const conditionMet = didDocument.service.some(
-          ({ claimType }) =>
```

Comment:
@jrhender or @JGiter is this really a good check if the condition is met? I feel like it should map through each condition and find it in the service array, not use .some on service array, right?

Reply:
(Asking because this logic was moved to claim-verification service, above.)

Reply:
@whitneypurdum It depends on `mustHaveAll` https://github.com/energywebfoundation/ew-credentials/blob/861feac6593b4807f1fc72e9f184357f7ac007fd/packages/credential-governance/contracts/profiles/enrolment-conditions/EnrolmentPrerequisiteRolesResolver.sol#L13

Reply:
@JGiter I still don't quite get it. This task is to ensure that the user 1. has the pre-requisite roles needed to take on a new role and 2. that all of those roles are valid. Should we change this so that this makes sure that every pre-requisite role that is needed is in the DID document?

Reply:
@whitneypurdum the enrollment preconditions are set in role definition. Depending on `mustHaveAll` preconditions are met when either one or all conditions are met. So the `.some` on service array should be used when `mustHaveAll` is `false` and `.every` should be otherwise

Reply:
OKay, thanks, I will leave as is then.